DESIGN_PRINCIPLES.md

```diff
@@ -0,0 +1,32 @@
+# Design principles
+
+This documents aims to capture the design principles that went into this library.
+It should serve as a reference point when making decisions on what features to include or exclude.
+
+## Simple
+
+One of most important goals that we want to adhere to is creating a _simple_ API.
+Overall, this means keeping the API as small as possible to get the task done.
+When in doubt, we'd rather not add flags or configuration options for certain use cases.
```

Comment:
This is a pretty big point, we like to emphasize for the Testcontainers project as a whole. Testcontainers does not intend to be a generic Docker client or a replacement for the Docker CLI. Instead, we follow an opinionated approach of providing APIs that make using best practices for integration testing easier. 

While it is technically possible to map to fixed ports, or specify fixed container names, we believe that this feature is actually not necessary for the intended use cases of Testcontainers, hence we would hesitate to just add all Docker capabilities to the Testcontainers API, without a good reason for the integration testing context.